Sources/ReleaseServer/ReleaseServer.WebApi/Common/JsonSerializable.cs

```diff
@@ -5,7 +5,7 @@ namespace ReleaseServer.WebApi.Common
 {
     public class JsonSerializable<T>
     {
-        public T FromJsonFile(string filePath)
+        public static T FromJsonFile(string filePath)
         {
             return JsonConvert.DeserializeObject<T>(File.ReadAllText(filePath));
```

Comment:
As you are using custom converters and the `JsonSerializerSettings` that are set for the ASPNET MVC will maybe be modified in the future, you should create a static class `JsonConfiguration` that can be used to centrally control these settings and pass them to every instance of a `JsonSerializer` and every call of `SerializeObject` and `DeserializeObject`:

```cs
public static class JsonConfiguration
{
    public static JsonSerializerSettings Settings
    {
        get
        {
            var settings = new JsonSerializerSettings();
            settings.Converters.Add(new ProductVersionConverter());
            return settings;
        }
    }
}
```

Usage:

```cs
public static FromJsonFile(string filePath)
{
    return JsonConvert.DeserializeObject<T>(File.ReadAllText(filePath), JsonConfiguration.Settings);
}
```

Reply:
Done, see https://github.com/Traeger-GmbH/release-server/pull/59/commits/c3ae5d6bbc3f460aa286d9cd83c27eeb0cc08606

Do i also have modify the Newtonsoft settings?
```csharp            
services
   .AddControllers()
   .AddNewtonsoftJson(options =>
   {
     options.SerializerSettings.Converters.Add(new ProductVersionConverter());
    });
```
If not, i have to modify the settings in two places, after adding a new custom json converter.

Is there any oportunity to consolidate the `NewtonsoftJson` settings with the ``JsonSerializerSettings`?

Reply:
Yes, to benefit from the central definition of these settings you should in fact use it as settings for the Newtonsoft Json instance used for the MVC.

The opportunity of "consolidating" the settings (say "use the same settings for every serialization task") is to have a single point of truth that describes how to serialize to and deserialize from JSON and thus the reduction of complexity.

Reply:
I implemented a solution that iterates threw the custom JsonSerializerSettings and adds the containing custom converters, see https://github.com/Traeger-GmbH/release-server/pull/59/commits/d3b0406d0f0acf4e93c6fb35f24934d50344e3a6

Reply:
To speed things up I implemented the changes necessary to obtain equivalent SerializerSettings.